modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -502,18 +504,40 @@ public void visit(BLangBinaryExpr binaryExpr) {
         resultTypes = types.checkTypes(binaryExpr, Lists.of(actualType), expTypes);
     }
 
+    public void visit(BLangTypeAccessExpr accessExpr) {
+        BType actualType = symTable.typeType;
+        accessExpr.resolvedType = symResolver.resolveTypeNode(accessExpr.typeNode, env);
+        resultTypes = types.checkTypes(accessExpr, Lists.of(actualType), expTypes);
+    }
+
     public void visit(BLangUnaryExpr unaryExpr) {
-        BType exprType = checkExpr(unaryExpr.expr, env).get(0);
```

Comment:
I think we can refactor this method. We check for (OperatorKind.TYPEOF.equals(unaryExpr.operator) twice in this method.

Reply:
+1